### PR TITLE
Output private key encrypted with password

### PR DESCRIPTION
**Pull request by paulfurley**
* Rewrite PgpKey class
  -  make PgpKey extend openpgp.Entity
  -  remove PgpKey.PublicKey
  -  add Armor and ArmorPrivate methods
  -  pass errors back up from PgpKey.Generate()

* Use github.com/fluidkeys/crypto fork

### CODE REVIEW

File: fluidkeys/main.go [hunk 1]
```diff
@@ -31,9 +31,14 @@ func (d DicewarePassword) AsString() string {
 	return strings.Join(d.words, d.separator)
 }
 
+type job struct {
```
Comment (idrysdale) [inline fluidkeys/main.go]: Could we better define `job`? Is it `generatePgpKey`?

Comment (idrysdale) [reply fluidkeys/main.go]: Ah no - there's a function called that.... 🤔

File: pgpkey/pgpkey.go [hunk 2]
```diff
@@ -17,25 +16,61 @@ const (
...
+	openpgp.Entity
 }
 
-func Generate(email string) PgpKey {
-	config := &packet.Config{RSABits: 4096}
+func Generate(email string) (*PgpKey, error) {
+	return generateKeyOfSize(email, RsaSizeSecureKeyBits)
+}
+
+func GenerateInsecure(email string) (*PgpKey, error) {
```
Comment (idrysdale) [inline pgpkey/pgpkey.go]: Can we make this private?

File: fluidkeys/main.go (hunk 1 above)
Comment (paulfurley) [reply fluidkeys/main.go]: renamed to `generatePgpKeyResult`

File: pgpkey/pgpkey.go (hunk 2 above)
Comment (paulfurley) [reply pgpkey/pgpkey.go]: Done

**Pull request merged**